helper-cli/src/main/kotlin/commands/VerifySourceArtifactCurationsCommand.kt

```diff
@@ -0,0 +1,127 @@
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.helper.commands
+
+import com.beust.jcommander.JCommander
+import com.beust.jcommander.Parameter
+import com.beust.jcommander.Parameters
+
+import com.here.ort.CommandWithHelp
+import com.here.ort.analyzer.HTTP_CACHE_PATH
+import com.here.ort.model.PackageCuration
+import com.here.ort.model.readValue
+import com.here.ort.utils.OkHttpClientHelper
+import com.here.ort.utils.PARAMETER_ORDER_MANDATORY
+import com.here.ort.utils.collectMessagesAsString
+import com.here.ort.utils.hash
+
+import java.io.File
+import java.io.IOException
+
+import okhttp3.Request
+
+import okio.buffer
+import okio.sink
+
+@Parameters(
+    commandNames = ["verify-source-artifact-curations"],
+    commandDescription = "Verifies that all curated source artifacts can be downloaded and that the hashes are correct."
+)
+internal class VerifySourceArtifactCurationsCommand : CommandWithHelp() {
+    @Parameter(
+        description = "A YAML file that contains package curation data.",
+        names = ["--package-curations-file"],
+        order = PARAMETER_ORDER_MANDATORY
+    )
+    private lateinit var packageCurationsFile: File
+
+    override fun runCommand(jc: JCommander): Int {
+        var success = true
+        val curations = packageCurationsFile.readValue<List<PackageCuration>>()
+
+        curations.forEach { curation ->
+            curation.data.sourceArtifact?.let { sourceArtifact ->
+                println("\n-----")
+                println("Checking source artifact for ${curation.id.toCoordinates()}.")
+                println("Downloading ${sourceArtifact.url}.")
+                val file = try {
+                    download(sourceArtifact.url)
+                } catch (e: Exception) {
+                    println("Failed to download source artifact: ${e.collectMessagesAsString()}")
+                    success = false
+                    null
+                }
+                if (file == null) {
+                    println("Aborting verification.")
+                } else {
+                    val hash = file.hash(sourceArtifact.hash.algorithm.aliases.first())
+                    println("Expected hash: ${sourceArtifact.hash.algorithm} ${sourceArtifact.hash.value}")
+                    println("Actual hash  : ${sourceArtifact.hash.algorithm} $hash")
+                    val verified = sourceArtifact.hash.verify(file)
+                    if (!verified) {
+                        println("Hashes do NOT match!")
+                        success = false
+                    }
+                }
+            }
+        }
+
+        println("\n-----")
+        return if (success) {
+            println("Successfully verified all source artifact curations.")
+            0
+        } else {
+            println("Some source artifact curations could NOT be verified, check the log for details.")
```

Comment:
Instead of saying just some, how about printing a summary like e.g.:

Verified N source artifact curations with M failures.
Failed:
-list of failed

Reply:
Done.